modules/network/kademlia/kademlia.js

```diff
@@ -431,6 +413,38 @@ class Kademlia {
             }
         });
 
+        this.node.use('*', (request, response, next) => {
+            if (request.params.header) {
+                const header = JSON.parse(request.params.header);
+                const destContact = header.to;
+                header.ttl -= 1;
+                if (header.ttl > 0) {
+                    this.log.info(`Request received for ${destContact}`);
+                    if (destContact === this.node.identity.toString('hex').toLowerCase()) {
```

Comment:
Do we have this identity in config in right format, so that we don't have to do this casting each time request is received?

Reply:
Replaced